### ISSUE TITLE
Don't list all of the keywords of Python and suggest them when I type a left square bracket

### ISSUE THREAD

**Issue by CodeCrazy-ywt** [microsoft/pylance-release#1712]
## Environment data

-   Language Server version: 2021.8.2
-   OS and version: Windows 10 20H2
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.6

## Expected behaviour

Don't list all of the keywords of Python and suggest them when I type a left square bracket

## Actual behaviour

Lists all of the keywords of Python and suggests them when I type a left square bracket.

## Logs

```
[FG] completion at c:\Users\Admin\ywt\test.py:0:1 ...
[FG]   parsing: c:\Users\Admin\ywt\test.py (6ms)
[FG]   parsing: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 0ms] (56ms)
[FG]   binding: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\builtins.pyi (13ms)
[FG]   binding: c:\Users\Admin\ywt\test.py (0ms)
[FG] completion at c:\Users\Admin\ywt\test.py:0:1 [found 191 items] (78ms)
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
[FG] parsing: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\typing.pyi [fs read 0ms] (15ms)
[FG] binding: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\typing.pyi (5ms)
[FG] parsing: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (10ms)
[FG] binding: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (1ms)
[FG] parsing: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi [fs read 0ms] (2ms)
[FG] binding: c:\Users\Admin\.vscode\extensions\ms-python.vscode-pylance-2021.8.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi (0ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629710631950 at c:\Users\Admin\ywt\test.py ...
[BG(1)]   parsing: c:\Users\Admin\ywt\test.py (0ms)
[BG(1)]   binding: c:\Users\Admin\ywt\test.py (1ms)
[BG(1)] getSemanticTokens delta previousResultId:1629710631950 at c:\Users\Admin\ywt\test.py (1ms)
Background analysis message: analyze
[BG(1)] analyzing: c:\Users\Admin\ywt\test.py ...
[BG(1)]   checking: c:\Users\Admin\ywt\test.py (5ms)
[BG(1)] analyzing: c:\Users\Admin\ywt\test.py (5ms)
Background analysis message: resumeAnalysis
Background analysis message: markFilesDirty
Background analysis message: analyze
```

## Code Snippet / Additional information

```python
XXX
```
![屏幕截图 2021-08-23 172237](https://user-images.githubusercontent.com/62426149/130424422-fadfe320-aeaa-4051-a777-8e2b19fc9b9f.png)

**Comment by CodeCrazy-ywt** [comment]
In addition, previously suggested items are also included in the suggestion list.

**Comment by jakebailey** [comment]
Pylance doesn't have contextual completions for keywords; we offer them up most everywhere, but with a very low priority compared to other items in the completion list.

Are you seeing otherwise? What do you mean by "previously suggested items are also included in the suggestion list"?

**Comment by jakebailey** [comment]
(See also #1557, which should be retitled to be more clear.)

**Comment by CodeCrazy-ywt** [comment]
> Are you seeing otherwise?

No, I haven't seen it elsewhere, I just see it when I type `[`(but not after a variable whose type is `dict`).

> What do you mean by "previously suggested items are also included in the suggestion list"?

Here's a screenshot. `min`, `max` and `print` are suggestions I previously accepted. 
![屏幕截图 2021-08-24 105626](https://user-images.githubusercontent.com/62426149/130549425-4e9b411d-fa2f-45dd-99f6-e4124981863d.png)

**Comment by erictraut** [comment]
This is what I'd expect to see in this case — at least given the current implementation. The names `min`, `max` and `print` are symbols that are used in the current scope, so they appear on the top of the list. As @jakebailey mentioned, keywords are currently suggested even in contexts where they are not appropriate, but they're always sorted low on the list. 

Perhaps as a simple short-term improvement we could omit all keyword suggestions if you haven't started to type an identifier. Once one an alphabetical character has been typed, we can suggest only the keywords that match that filter. For example, if you were to place the insertion point between the square brackets, no keywords would be suggested until you type a character, say an `n`, and then the keyword `not` would be included in the suggestion list. @jakebailey, if you like that idea, it's a trivial change — just change [this line](https://github.com/microsoft/pyright/blob/3dd4e431d90038c9755c04e3ad034274fe2efcba/packages/pyright-internal/src/languageService/completionProvider.ts#L1838) to return false.

In the future, we could add further smarts to offer only the keywords that are syntactically appropriate at the current location, but that's a much larger change.
